src/components/ui/DeviceMotionView.tsx

```diff
@@ -1,79 +1,94 @@
-import { Text, View } from "react-native";
-import React, { useState, useEffect } from "react";
-import { DeviceMotion } from "expo-sensors";
+import React, { useEffect } from "react";
+
+import Stack from "@src/components/ui/Stack";
+import { useUser } from "@src/state/useUser";
 import * as Haptics from "expo-haptics";
+import { DeviceMotion } from "expo-sensors";
+import { Switch, Text } from "react-native";
 
-export default function DeviceMotionView(props: any) {
-  const [{ beta, gamma }, setRotationData] = useState({ beta: 0, gamma: 0 });
-  const [orientation, setOrientation] = useState(0);
-  const [notification, setNotification] = useState(false);
-  const showNotification = () => setNotification(true);
-  const hideNotification = () => setNotification(false);
-  const { isTrackingEnabled } = props;
+export default function DeviceMotionView() {
+  const isTrackingEnabled = useUser((state) => state.isTrackingEnabled);
+  const setTrackingEnabled = useUser((state) => state.setTrackingEnabled);
 
-  //Get RotationData
-  useEffect(() => {
-    if (isTrackingEnabled) {
-      _subscribe();
-    } else {
-      hideNotification();
-    }
-    return () => {
-      _unsubscribe();
-    };
-  }, [isTrackingEnabled]);
+  const setCurrentPosture = useUser((state) => state.setCurrentPosture);
+  const mode = useUser((state) => state.mode);
 
-  //Get rotation and orientation values
-  const _subscribe = async () => {
-    try {
-      DeviceMotion.addListener((devicemotionData) => {
-        if (devicemotionData.rotation) {
-          setRotationData(devicemotionData.rotation);
-        }
-        setOrientation(devicemotionData.orientation);
-      });
+  const toggleTracking = () => {
+    const value = !isTrackingEnabled;
 
-      DeviceMotion.setUpdateInterval(1000);
-    } catch (error) {
-      console.error(error);
+    if (!value) {
+      setCurrentPosture("not_reading");
+    } else {
+      DeviceMotion.requestPermissionsAsync();
     }
+    setTrackingEnabled(value);
   };
 
-  const _unsubscribe = () => {
-    DeviceMotion.removeAllListeners();
-  };
-
-  //Provide feedback
   useEffect(() => {
-    hideNotification();
+    const _subscribe = async () => {
+      try {
+        DeviceMotion.addListener((devicemotionData) => {
+          try {
+            // Given acceleration including gravity components
+            const a_y = devicemotionData?.accelerationIncludingGravity?.y;
+            const a_z = devicemotionData?.accelerationIncludingGravity?.z;
+            // Function to convert radians to degrees
+            function radiansToDegrees(radians: number) {
+              return radians * (180 / Math.PI);
+            }
+            // Calculate pitch in radians
+            const pitchRadians = Math.atan2(a_y, a_z);
+            // Convert pitch to degrees
+            const pitchDegrees = radiansToDegrees(pitchRadians);
+            if (pitchDegrees < -120 && pitchDegrees > -179) {
```

Comment:
It would be nice to have the range for each posture type configurable, in cases we need to change them for testing. But overall it works great!